scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5159 @@
+        -----
+        Suppose continuous probability distribution has support :math:`[l, r]`.
+        By definition of the support, the PDF evaluates to its minimum value
+        of :math:`0` outside the support; i.e. for :math:`x < l` or
+        :math:`x > r`. The maximum of the PDF may be less than or greater than
+        :math:`1`; since the valus is a probability *density*, only its integral
+        over the support must equal :math:`1`.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1., b=1.)
+
+        Evaluate the probability density function at the desired argument:
+
+        >>> X.pdf(0.25)
+        0.5
+
+        """
+        return self._pdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _pdf_dispatch(self, x, *, method=None, **kwargs):
+        if self._overrides('_pdf_formula'):
+            method = self._pdf_formula
+        else:
+            method = self._pdf_logexp
+        return method
+
+    def _pdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _pdf_logexp(self, x, **kwargs):
+        return np.exp(self._logpdf_dispatch(x, **kwargs))
+
+    ## Cumulative Distribution Functions
+
+    def logcdf(self, x, y=None, *, method=None):
+        r"""Log of the cumulative distribution function
+
+        The cumulative distribution function :math:`F(x)` is the probability
+        the random variable :math:`X` will take on a value less than or equal
+        to :math:`x`:
+
+        .. math::
+
+            F(x) = P(X ≤ x)
+
+        A two-argument variant of this function is also defined as the
+        probability the random variable :math:`X` will take on a value between
+        :math:`x` and :math:`y`.
+
+        .. math::
+
+            F(x, y) = P(x ≤ X ≤ y)
+
+        `logcdf` computes the logarithm of the cumulative distribution function,
+        :math:`\log(F(x))`/:math:`\log(F(x, y))`, but it may be numerically
+        favorable compared to the naive implementation (computing the CDF
+        and taking the logarithm).
+
+        `logcdf` accepts `x` for :math:`x` and `y` for :math:`y`.
+
+        Parameters
+        ----------
+        x, y : Array
+            The arguments of the log of the cumulative distribution function
+            (log-CDF). `x` is required; `y` is optional.
+        method : {None, 'formula', 'logexp', 'complement', 'quadrature', 'subtraction'}
+            The strategy used to evaluate the log-CDF.
+            By default (``None``), the one-argument form of the function
+            chooses between the following options, listed in order of precedence.
+
+            - ``'formula'``: use a formula for the log-CDF itself
+            - ``'logexp'``: evaluate the CDF directly and take the logarithm
+            - ``'complement'``: evaluate the logarithm of the complementary CDF
```

Comment:
A formula would be good here.

Reply:
Where? Every place *complement* is used, or just here, or just where it says "logarithmic complement"? Do you want the conceptual formula, or the actual implementation? Can this be in a tutorial that describes the numerical methods instead of in every docstring, since the idea of "complement" is the same in every method?

Reply:
A conceptual illustration, a formula or pseudo code would do. Tutorial is good, but there's a one-time reader.

Reply:
How about referring the reader to the notes and including:

        The "logarithmic complement" of :math:`z` is mathematically equivalent to
         :math:`\log(1-\exp(z))`, but it is computed to avoid loss of precision
         when :math:`\exp(z)` is nearly :math:`0` or :math:`1`. 

The implementation is:

```python3
    def f1(x):
        # good for exp(x) close to 0
        return np.log1p(-np.exp(x))

    def f2(x):
        # good for exp(x) close to 1
        return np.real(np.log(-special.expm1(x + 0j)))

    return _lazywhere(x < -1, (x,), f=f1, f2=f2)[()]
```
so I'd rather not include that here.

Reply:
Just `log(cdf(1-p))` would do. Literally, after "evaluate the logatithm of the compementary CDF, `log(cdf(1-p))`".

Reply:
I think you mean `log(1 - exp(logccdf(x)))`, but this information is now in the notes.

Reply:
The very fact that I got it wrong is IMO a strong indicator that this info would be very helpful to duplicate right here, in the form you wrote above. 
In addition to what's in the notes (which is great)